### ISSUE TITLE
E tensorflow/tsl/lib/monitoring/collection_registry.cc:81] Cannot register 2 metrics with the same name: /tensorflow/core/bfc_allocator_delay

### ISSUE THREAD

**Issue by tomjrtsmith** [tensorflow/tensorflow#59419]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Have you reproduced the bug with TF nightly?

No

### Source

source

### Tensorflow Version

2.11.0

### Custom Code

Yes

### OS Platform and Distribution

Linux Ubuntu 22.04

### Python version

3.10

### Current Behaviour?

```shell
Using Tensorflow with Freqtrade, seems to be occurring on a few installs using some custom coding. I've tracked down the class I think it is coming from.
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import numpy as np

# Restricted Boltzmann Machine

class RBM():
    def __init__(self, nv=28 * 28, nh=512, cd_steps=3):
        tf.random.set_seed(42)
        self.graph = tf.Graph()
        with self.graph.as_default():
            tf.random.set_seed(42)
            self.W = tf.Variable(tf.compat.v1.truncated_normal((nv, nh)) * 0.01)
            self.bv = tf.Variable(tf.zeros((nv, 1)))
            self.bh = tf.Variable(tf.zeros((nh, 1)))

            self.cd_steps = cd_steps
            self.modelW = None

    def bernoulli(self, p):
        return tf.nn.relu(tf.sign(p - tf.compat.v1.random_uniform(p.shape)))

    def energy(self, v):
        b_term = tf.matmul(v, self.bv)
        linear_tranform = tf.matmul(v, self.W) + tf.squeeze(self.bh)
        h_term = tf.reduce_sum(tf.compat.v1.log(tf.exp(linear_tranform) + 1), axis=1)
        return tf.reduce_mean(-h_term - b_term)

    def sample_h(self, v):
        ph_given_v = tf.sigmoid(tf.matmul(v, self.W) + tf.squeeze(self.bh))
        return self.bernoulli(ph_given_v)

    def sample_v(self, h):
        pv_given_h = tf.sigmoid(tf.matmul(h, tf.transpose(self.W)) + tf.squeeze(self.bv))
        return self.bernoulli(pv_given_h)

    def gibbs_step(self, i, k, vk):
        hk = self.sample_h(vk)
        vk = self.sample_v(hk)
        return i + 1, k, vk

    def train(self, X, lr=0.01, batch_size=64, epochs=5):
        with self.graph.as_default():
            tf_v = tf.compat.v1.placeholder(tf.float32, [batch_size, self.bv.shape[0]])
            v = tf.round(tf_v)
            vk = tf.identity(v)

            i = tf.constant(0)
            _, _, vk = tf.while_loop(cond=lambda i, k, *args: i <= k,
                                     body=self.gibbs_step,
                                     loop_vars=[i, tf.constant(self.cd_steps), vk],
                                     parallel_iterations=1,
                                     back_prop=False)

            vk = tf.stop_gradient(vk)
            loss = self.energy(v) - self.energy(vk)
            optimizer = tf.compat.v1.train.AdamOptimizer(lr).minimize(loss)
            init = tf.compat.v1.global_variables_initializer()

        with tf.compat.v1.Session(graph=self.graph) as sess:
            init.run()
            for epoch in range(epochs):
                losses = []
                for i in range(0, len(X) - batch_size, batch_size):
                    x_batch = X[i:i + batch_size]
                    l, _ = sess.run([loss, optimizer], feed_dict={tf_v: x_batch})
                    losses.append(l)
                print('Epoch Cost %d: ' % epoch, np.mean(losses), end='\r')
            self.modelW = self.W.eval()
```

### Relevant log output

```shell
E tensorflow/tsl/lib/monitoring/collection_registry.cc:81] Cannot register 2 metrics with the same name: /tensorflow/core/bfc_allocator_delay
/home/ubuntu/freqtrade/.env/lib/python3.10/site-packages/tensorflow_io/python/ops/__init__.py:98: UserWarning: unable to load libtensorflow_io_plugins.so: unable to open file: libtensorflow_io_plugins.so, from paths: ['/home/ubuntu/freqtrade/.env/lib/python3.10/site-packages/tensorflow_io/python/ops/libtensorflow_io_plugins.so']
caused by: ['/home/ubuntu/freqtrade/.env/lib/python3.10/site-packages/tensorflow_io/python/ops/libtensorflow_io_plugins.so: undefined symbol: _ZN3tsl6Status22MaybeAddSourceLocationENS_14SourceLocationE']
  warnings.warn(f"unable to load libtensorflow_io_plugins.so: {e}")
/home/ubuntu/freqtrade/.env/lib/python3.10/site-packages/tensorflow_io/python/ops/__init__.py:104: UserWarning: file system plugins are not loaded: unable to open file: libtensorflow_io.so, from paths: ['/home/ubuntu/freqtrade/.env/lib/python3.10/site-packages/tensorflow_io/python/ops/libtensorflow_io.so']
caused by: ['/home/ubuntu/freqtrade/.env/lib/python3.10/site-packages/tensorflow_io/python/ops/libtensorflow_io.so: undefined symbol: _ZTVN10tensorflow13GcsFileSystemE']
  warnings.warn(f"file system plugins are not loaded: {e}")
unknown 2.12.0-dev20230120
```
</details>

**Comment by tomjrtsmith** [comment]
this I believe is the code that is causing the error

# class that implements an Anomaly detector using the Isolation Forest (IFOR) algorithm


import numpy as np
from pandas import DataFrame, Series
import pandas as pd

pd.options.mode.chained_assignment = None  # default='warn'

# Strategy specific imports, files must reside in same folder as strategy
import sys
from pathlib import Path

sys.path.append(str(Path(__file__).parent))

import logging
import warnings

log = logging.getLogger(__name__)
# log.setLevel(logging.DEBUG)
warnings.simplefilter(action='ignore', category=pd.errors.PerformanceWarning)

import random

import os

os.environ['TF_CPP_MIN_LOG_LEVEL'] = '1'
os.environ['TF_DETERMINISTIC_OPS'] = '1'

import tensorflow as tf

seed = 42
os.environ['PYTHONHASHSEED'] = str(seed)
random.seed(seed)
tf.random.set_seed(seed)
np.random.seed(seed)

tf.compat.v1.logging.set_verbosity(tf.compat.v1.logging.WARN)

import keras
from keras import layers
from sklearn.ensemble import IsolationForest
from ClassifierSklearn import ClassifierSklearn

import h5py
import joblib

class AnomalyDetector_IFOR(ClassifierSklearn):

    classifier = None
    clean_data_required = False # training data should not contain anomalies


    def create_classifier(self):
        classifier = IsolationForest(contamination=self.contamination)
        return classifier

**Comment by gaikwadrahul8** [comment]
Hi, @SuryanarayanaY 

Could you please look into this issue ? Thank you!

**Comment by tomjrtsmith** [comment]
Hi, is there any way to diagnose this error? It seems to be persistent, we've removed unused tensorflow random seed references from code and it still occurs.

**Comment by SuryanarayanaY** [comment]
Hi @tomjrtsmith ,
Could you please confirm whether the error reported by you observed in TF2.11 version or tf-nightly ? You mentioned TF version as 2.11 but in log stack I can see `2.12.0-dev20230120` indicating it as nightly version.
We have observed this error log in nightly version but not in TF 2.11v and a similar issue #58889 already bought into developer notice and the team is working on it. AFAIK even though it logs under Error but seems it's not affecting functionality as confirmed by the user also in mentioned ref issue.
Please confirm whether the Error log from nightly or 2.11v and also whether it is affecting functionality ?
Thank you!

**Comment by tomjrtsmith** [comment]
Yes this is affecting functionality, i.e.the application will not run. I
have deprecated the numpy version to 1.23.5 as there is a pip dependency
error with tensorflow.

The application it is running in is freqtrade
https://github.com/freqtrade/freqtrade

The two strategies that are causing the error are

https://github.com/nateemma/strategies/blob/master/binanceus/Anomaly_nseq.py
and
https://github.com/nateemma/strategies/blob/master/binanceus/PCA_nseq.py

They are erroring when running on ARM chip VPS's. Other strategies using
Tensorflow will run ok. It is just these two.

Tensorflow version is 2.11 as shown below.

pip show tensorflow
Name: tensorflow
Version: 2.11.0
Summary: TensorFlow is an open source machine learning framework for
everyone.
Home-page: https://www.tensorflow.org/
Author: Google Inc.
Author-email: ***@***.***
License: Apache 2.0
Location: /home/ubuntu/freqtrade/.env/lib/python3.10/site-packages
Requires: tensorflow-cpu-aws

**Comment by SuryanarayanaY** [comment]
@tomjrtsmith ,
Thanks for confirmation regarding your ARM chip. Please refer to this [note](https://www.tensorflow.org/install/pip#:~:text=Nightly-,Note%3A%20Starting%20with%20TensorFlow%202.10%2C%20Linux%20CPU%2Dbuilds%20for%20Aarch64,package.%20See%20this%20blog%20post%20for%20more%20information%20about%20this%20collaboration.,-conda%20install%20%2D) from Official documentation.

Anyways I tried it on a VM with ARM architecture and tried a simple code to test whether tensorflow running successfully or not and found that its actually working even though the reported error log by you also persists in my case also with tf-nightly only.With TF2.11v i don't find the error log mentioned by you as attached below and Iam sure this log you notice only with tf-nightly version as I can see in `your attached error log `I can see `unknown 2.12.0-dev20230120` indicates nightly version.

`E tensorflow/tsl/lib/monitoring/collection_registry.cc:81] Cannot register 2 metrics with the same name: /tensorflow/core/bfc_allocator_delay
`

Please refer below snapshot of log with TF2.11 version and the tested code output also without any error.

<img width="1503" alt="Screenshot 2023-01-30 at 3 04 35 PM" src="https://user-images.githubusercontent.com/116063290/215440499-7a9a0317-c9a6-4f71-818d-22c6b964f771.png">

Please refer below snapshot of log with TF2.12 version and the tested code output also success even though reported error persists in log but code executed without any problem here.

<img width="1498" alt="Screenshot 2023-01-30 at 2 40 24 PM" src="https://user-images.githubusercontent.com/116063290/215441397-184ad57f-041f-4db7-9faa-660eba3af04a.png">

I have doubt on your environment.From the error log you attached I see `unknown 2.12.0-dev20230120` which is strange.May be your project is looking for tf-nightly version = 2.12.0-dev20230120 which means its looking for nightly build of date 2023.01.20 but its not found actually. Can you print and confirm the output of `tf.__version__` to check which versions actually installed. The environment might be corrupted due to numerous pip install and pip uninstall commands.I recommend to create a fresh environment with TF 2.11 version and others required for your project.

**Comment by tomjrtsmith** [comment]
Interesting. I think the problem may lie in that Freqtrade has just
recently started using tensorflow as part of its base install and people
are developing strats on a few different architectures, I've got another
with memory bleed on a linux VM but works fine on MacOS, I didn't know
there was a cpu only Tensorflow version but I installed that on the other
instance and it seemed to help. Thanks Suryanarayana, I'll run some more
tests and report back. Thankyou!
